lib/phoenix_live_view/upload.ex

```diff
@@ -184,6 +184,19 @@ defmodule Phoenix.LiveView.Upload do
     |> update_uploads(socket)
   end
 
+  @doc """
+  Retrieves the `%UploadConfig{}` from the socket for the provided ref.
+
+  If the `socket` contains uploads and the given `ref` then the value is returned in the shape of `{:ok, upload_conf}`.
+  If the `socket` does not have any uploads allowed, or the `ref` is not found, `:error` is returned.
+  """
+  def get_upload_by_ref(%Socket{} = socket, config_ref) do
+    with {:ok, uploads} <- Map.fetch(socket.assigns, :uploads),
+         {:ok, name} <- Map.fetch(uploads[@refs_to_names], config_ref) do
+      Map.fetch(uploads, name)
+    end
```

Comment:
```suggestion
    with %{assigns: %{uploads: uploads}} <- socket,
         %{@refs_to_names => %{^config_ref => name}} <- uploads do
      Map.fetch(uploads, name)
    else
      _ -> :error
    end
```

And the `get_upload_by_ref!` can be similar rewritten to use pattern matching instead of calling `Map.fetch!` twice. :)